saleor/payment/gateways/dummy/__init__.py

```diff
@@ -104,21 +124,9 @@ def process_payment(
     """Process the payment."""
     token = payment_information.token
 
-    # Process payment normally if payment token is valid
-    if token not in dict(ChargeStatus.CHOICES):
-        return capture(payment_information, config)
-
-    # Process payment by charge status which is selected in the payment form
-    # Note that is for testing by dummy gateway only
-    charge_status = token
-    authorize_response = authorize(payment_information, config)
-    if charge_status == ChargeStatus.NOT_CHARGED:
-        return authorize_response
-
-    if not config.auto_capture:
-        return authorize_response
+    if token in PREAUTHORIZED_TOKENS:
+        authorize_response = authorize(payment_information, config)
+        if not config.auto_capture:
+            return authorize_response
```

Comment:
Just noticed this - here we always call `authorize`, but we could only call it when it's needed so when `auto_capture` is disabled. Something like that should work:

```suggestion
    if token in PREAUTHORIZED_TOKENS and not config.auto_capture:
        return authorize(payment_information, config)
```

Reply:
We decided to leave it as it is. Even if `auto_capture` is set to True, `authorize` should be performed for `PREAUTHORIZED_TOKENS`.